python/drned_xmnr/op/common_op.py

```diff
@@ -102,3 +110,66 @@ def perform(self):
         if self.filter.devcli_error is None:
             return {'success': 'Saved initial config.'}
         return {'failure': 'Could not save config.'}
+
+
+class ParseLogErrorsOp(ActionBase):
+    """ Action handler used to parse and print errors from drned-xmnr logfiles.
+    """
+    action_name = 'xmnr parse-log-errors'
+
+    def _init_params(self, params):
+        self.target_log = self.param_default(params, "target_log", None)
+
+    def perform(self):
+        return self.run_with_trans(self._perform, write=True)
+
+    def _perform(self, trans):
+        filepath = self._get_target_filepath()
+        try:
+            with open(filepath, 'r') as logfile:
+                problems = gather_problems(logfile)
+                self._store_problems(trans, problems)
+        except OSError as ose:
+            # TODO - do not reveal potential filepath attempted?
+            msg = "Couldn't read target log file: {0}".format(os.strerror(ose.errno))
+            raise ActionError(msg)
+
+        return {'success': 'Target %s parsed.' % self.target_log}
+
+    def _get_target_filepath(self):
+        tag = self.target_log
+        if tag == ns.drned_xmnr_common_xmnr_log:
+            path = os.path.join(
+                os.getcwd(), 'logs', 'ncs-python-vm-drned-xmnr.log'
+            )
+        elif tag == ns.drned_xmnr_device_trace:
+            path = os.path.join(
+                self.dev_test_dir,
+                'netconf-' + self.dev_name + '.trace'
```

Comment:
netconf trace file is located in logs too, it is not in the test dir, or is it?

Reply:
fixed